lib/validate.tsx

```diff
@@ -75,4 +84,23 @@ export const fundAccountSchemaClient = yup
 
 export const fundAccountSchemaServer = yup
   .object()
-  .shape(fundAccountSchemaServerShape)
+  .shape(fundAccountSchemaServerShape, [["h-captcha-response", "api-key"]])
+
+export function verifyAPIKey(req: string, keys: string[]): boolean {
+  const len = req.length
+  for (const key of keys) {
+    if (key.length != len) {
+      continue
+    }
+    // Do constant time comparison of the API key for protection
+    // against timing attacks.
+    let err = 0
+    for (let i = 0; i < len; ++i) {
+      err |= req.charCodeAt(i) ^ key.charCodeAt(i)
+    }
+    if (err == 0) {
+      return true
+    }
+  }
+  return false
+}
```

Comment:
I would move this function into `pages/api/account.ts` since it's not really a format validator.